aten/src/ATen/native/mps/operations/Indexing.h

```diff
@@ -0,0 +1,51 @@
+//  Copyright © 2022 Apple Inc.
+
+#include <ATen/ATen.h>
+#include <ATen/Tensor.h>
+#include <ATen/Utils.h>
+#include <ATen/mps/MPSStream.h>
+#include <ATen/native/mps/TensorFactory.h>
+#include <c10/core/ScalarType.h>
+#include <torch/library.h>
+#include <unordered_map>
+
+using namespace at::mps;
+
+namespace at {
+namespace native {
+namespace mps {
+
+std::string getMetalScalarType(ScalarType scalar_type) {
```

Comment:
Nit (or return `const std::string&`) <- i.e. please return string literal type to avoid unnecessary copy
```suggestion
const char* getMetalScalarType(ScalarType scalar_type) {
```